dev_scripts/env.py

```diff
@@ -0,0 +1,503 @@
+RUN cd /home/user/dangerzone && poetry install --no-root
+"""
+
+DOCKERFILE_BUILD_DEBIAN_DEPS = ""
+
+# FIXME: Drop this fix after it's resolved upstream.
+# See https://github.com/freedomofpress/dangerzone/issues/286#issuecomment-1347149783
+DOCKERFILE_BUILD_FEDORA_DEPS = r"""
+RUN dnf reinstall -y shadow-utils
+"""
+
+# The Dockerfile for building an environment with Dangerzone installed in it. Parts of
+# the Dockerfile will be populated during runtime.
+#
+# FIXME: The fact that we remove the package does not reduce the image size. We need to
+# flatten the image layers as well.
+DOCKERFILE_BUILD = r"""FROM {distro}:{version}
+
+{install_deps}
+
+COPY {package} /tmp/{package}
+RUN {install_cmd} /tmp/{package}
+RUN rm /tmp/{package}
+
+#########################################
+# Create a non-root user to run Dangerzone
+RUN adduser user
+# See https://github.com/freedomofpress/dangerzone/issues/286#issuecomment-1347149783
+RUN echo user:2000:2000 > /etc/subuid
+RUN echo user:2000:2000 > /etc/subgid
+
+USER user
+WORKDIR /home/user
+"""
+
+
+def run(*args):
+    """Simple function that runs a command, validates it, and returns the output"""
+    return subprocess.run(
+        args, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    ).stdout
+
+
+def git_root():
+    """Get the root directory of the Git repo."""
+    # FIXME: Use a Git Python binding for this.
+    # FIXME: Make this work if called outside the repo.
+    path = run("git", "rev-parse", "--show-toplevel").decode().strip("\n")
+    return pathlib.Path(path)
+
+
+def distro_root(distro, version):
+    """Get the root directory for the specific Linux environment."""
+    return git_root() / f"dev_scripts/envs/{distro}/{version}"
+
+
+def distro_state(distro, version):
+    """Get the directory where we will store the state for the distro."""
+    return distro_root(distro, version) / "state"
+
+
+def distro_build(distro, version):
+    """Get the directory where we will store the build files for the distro."""
+    return distro_root(distro, version) / "build"
+
+
+def image_name_build(distro, version):
+    """Get the container image for the dev variant of a Dangerzone environment."""
+    return IMAGE_NAME_BUILD_DEV_FMT.format(distro=distro, version=version)
+
+
+def image_name_install(distro, version):
+    """Get the container image for the Dangerzone environment."""
+    return IMAGE_NAME_BUILD_FMT.format(distro=distro, version=version)
+
+
+def dz_version():
+    """Get the current Dangerzone version."""
+    with open(git_root() / "share/version.txt") as f:
+        return f.read().strip()
+
+
+class Env:
+    """A class that implements actions on Dangerzone environments"""
+
+    def __init__(self, distro, version, runtime):
+        """Initialize an Env class based on some common parameters."""
+        self.distro = distro
+        self.version = version
+
+        # Try to autodetect the runtime, if the user has not provided it.
```

Comment:
I don't think we can do that, because the developer can just as well use Docker for the **environment** where they will run Dangerzone in. That environment will of course end up having Podman installed there, but that's different. So, what I'm doing here ("use Podman, else Docker, else fail") does not align with Dangerzone's needs.